File: backend/src/main/java/com/yangshengzhou/lucky_sms/pojo/Major.java
package com.yangshengzhou.lucky_sms.pojo;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;

/**
 * 专业实体类
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("majors")
public class Major extends BaseEntity {
    
    @TableId(type = IdType.AUTO)
    private Integer majorId;
    
    @TableField("major_code")
    private String majorCode;
    
    @TableField("major_name")
    private String majorName;
    
    @TableField("department_id")
    private Integer departmentId;
    
    @TableField("major_type_id")
    private Integer majorTypeId;
    
    @TableField("degree_type_id")
    private Integer degreeTypeId;
    
    @TableField("duration_years")
    private Integer durationYears;
    
    private String description;
    
    @TableField("status_id")
    private Integer statusId;
    

    @TableField(exist = false)
    private Department department;
    
    @TableField(exist = false)
    private StatusType status;
}

File: backend-go/models/major.go
package models

type Major struct {
	BaseEntity
	MajorID       int    `json:"majorId" gorm:"column:major_id;primaryKey;autoIncrement"`
	MajorCode     string `json:"majorCode" gorm:"column:major_code;size:20;uniqueIndex"`
	MajorName     string `json:"majorName" gorm:"column:major_name;size:100"`
	DepartmentID  int    `json:"departmentId" gorm:"column:department_id;index"`
	Description   string `json:"description" gorm:"column:description;type:text"`
	StatusID      int    `json:"statusId" gorm:"column:status_id"`
	
	Department *Department `json:"department,omitempty" gorm:"-"`
	Status     *StatusType `json:"status,omitempty" gorm:"-"`
}

func (Major) TableName() string {
	return "majors"
}
